lib/solarwinds_apm.rb

```diff
@@ -5,69 +5,73 @@
 # Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
 
 begin
+  require 'solarwinds_apm/version'
+  require 'solarwinds_apm/logger'
   if ENV.fetch('SW_APM_ENABLED', 'true') == 'false'
     SolarWindsAPM.logger.warn 'SW_APM_ENABLED environment variable detected and was set to false; SolarWindsAPM disabled'
     return
   end
 
-  require 'solarwinds_apm/version'
-  require 'solarwinds_apm/thread_local'
-  require 'solarwinds_apm/support_report'
-  require 'solarwinds_apm/constants'
-  require 'solarwinds_apm/api'
-  require 'solarwinds_apm/base'
-  require 'solarwinds_apm/logger'
-  require 'solarwinds_apm/config'
-
-  SolarWindsAPM::Config.load_config_file
-  SolarWindsAPM.loaded = false
   begin
     if /linux/.match?(RUBY_PLATFORM)
-      require_relative './libsolarwinds_apm.so'
-      require 'solarwinds_apm/oboe_init_options'
-      require_relative './oboe_metal'  # initialize Reporter; sets SolarWindsAPM.loaded = true if successful
+      require 'solarwinds_apm/config'
+      SolarWindsAPM::Config.load_config_file
+
+      require_relative './libsolarwinds_apm.so'       # load c-lib oboe
+      require 'solarwinds_apm/oboe_init_options'      # setup oboe reporter options
+      require_relative './oboe_metal'                 # initialize reporter: SolarWindsAPM.loaded = true
+      
+      SolarWindsAPM.logger.info '==================================================================='
+      SolarWindsAPM.logger.info "SolarWindsAPM info: Platform #{RUBY_PLATFORM}."
```

Comment:
Can we also add a log message about the Ruby version?

Reply:
added